packages/styled/types/base.d.ts

```diff
@@ -13,19 +17,22 @@ export {
 
 export { ComponentSelector, Interpolation }
 
-/** Same as StyledOptions but shouldForwardProp must be a type guard */
+/**
+ * Same as StyledOptions but shouldForwardProp must be a type guard (https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates).
```

Comment:
I don't think this link to the TS handbook is necessary, and it will eventually become outdated if they release a new version of the handbook.